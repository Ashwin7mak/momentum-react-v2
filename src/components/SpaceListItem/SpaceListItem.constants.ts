import { TEAM_COLORS } from '../ThemeProvider/ThemeProvider.constants';

const CLASS_PREFIX = 'md-space-list-item';

const DEFAULTS = {
  AVATAR_PROPS: { title: 'C' },
  TEAM_COLOR: TEAM_COLORS.default,
};

const STYLE = {
  textWrapper: `${CLASS_PREFIX}-text-wrapper`,
  isNewActivity: `${CLASS_PREFIX}-is-new-activity`,
  dotDivider: `${CLASS_PREFIX}-dot-divider`,
};

export { CLASS_PREFIX, DEFAULTS, STYLE };
