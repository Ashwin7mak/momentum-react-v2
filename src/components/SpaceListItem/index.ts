import { default as SpaceListItem } from './SpaceListItem';
import * as CONSTANTS from './SpaceListItem.constants';
import { Props } from './SpaceListItem.types';

export { CONSTANTS as SPACE_LIST_ITEM_CONSTANTS };

export type SpaceListItemProps = Props;

export default SpaceListItem;
