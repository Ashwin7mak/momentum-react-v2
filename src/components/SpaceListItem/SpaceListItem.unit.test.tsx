import React from 'react';

import SpaceListItem from './';
import { TEAM_COLORS } from '../ThemeProvider/ThemeProvider.constants';
import { TeamColor } from '../ThemeProvider/ThemeProvider.types';
import Avatar from '../Avatar';
import Icon from '../Icon';
import { mountAndWait } from '../../../test/utils';

describe('<SpaceListItem />', () => {
  describe('snapshot', () => {
    it('should match snapshot', async () => {
      expect.assertions(1);

      const container = await mountAndWait(<SpaceListItem />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with className', async () => {
      expect.assertions(1);

      const className = 'example-class';

      const container = await mountAndWait(<SpaceListItem className={className} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with id', async () => {
      expect.assertions(1);

      const id = 'example-id';

      const container = await mountAndWait(<SpaceListItem id={id} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with style', async () => {
      expect.assertions(1);

      const style = { color: 'pink' };

      const container = await mountAndWait(<SpaceListItem style={style} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with firstLine', async () => {
      expect.assertions(1);

      const firstLine = 'firstLine';

      const container = await mountAndWait(<SpaceListItem firstLine={firstLine} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with secondLine', async () => {
      expect.assertions(1);

      const secondLine = 'secondLine';

      const container = await mountAndWait(<SpaceListItem secondLine={secondLine} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with isNewActivity', async () => {
      expect.assertions(1);

      const isNewActivity = true;

      const container = await mountAndWait(<SpaceListItem isNewActivity={isNewActivity} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with teamColor', async () => {
      expect.assertions(1);

      const teamColor = TEAM_COLORS.cobalt;

      const container = await mountAndWait(<SpaceListItem teamColor={teamColor as TeamColor} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with isMention', async () => {
      expect.assertions(1);

      const isMention = true;

      const container = await mountAndWait(<SpaceListItem isMention={isMention} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with isEnterRoom', async () => {
      expect.assertions(1);

      const isEnterRoom = true;

      const container = await mountAndWait(<SpaceListItem isEnterRoom={isEnterRoom} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with isAlertMuted', async () => {
      expect.assertions(1);

      const isAlertMuted = true;

      const container = await mountAndWait(<SpaceListItem isAlertMuted={isAlertMuted} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with isError', async () => {
      expect.assertions(1);

      const isError = true;

      const container = await mountAndWait(<SpaceListItem isError={isError} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with action', async () => {
      expect.assertions(1);

      const action = <p>action</p>;

      const container = await mountAndWait(<SpaceListItem action={action} />);

      expect(container).toMatchSnapshot();
    });

    /* ...additional snapshot tests... */
  });

  describe('attributes', () => {
    it('should have provided class when className is provided', async () => {
      expect.assertions(1);

      const className = 'example-class';

      const element = (await mountAndWait(<SpaceListItem className={className} />))
        .find(SpaceListItem)
        .getDOMNode();

      expect(element.classList.contains(className)).toBe(true);
    });

    it('should have provided id when id is provided', async () => {
      expect.assertions(1);

      const id = 'example-id';

      const element = (await mountAndWait(<SpaceListItem id={id} />))
        .find(SpaceListItem)
        .getDOMNode();

      expect(element.id).toBe(id);
    });

    it('should have provided style when style is provided', async () => {
      expect.assertions(1);

      const style = { color: 'pink' };
      const styleString = 'color: pink;';

      const element = (await mountAndWait(<SpaceListItem style={style} />))
        .find(SpaceListItem)
        .getDOMNode();

      expect(element.getAttribute('style')).toBe(styleString);
    });

    it('should have provided avatarProps when id is avatarProps', async () => {
      expect.assertions(1);

      const avatarProps = { title: 'C' };

      const element = (await mountAndWait(<SpaceListItem avatarProps={avatarProps} />))
        .find(Avatar)
        .find('span')
        .getDOMNode();

      expect(element.textContent).toBe(avatarProps.title);
    });

    it('should have provided firstLine when firstLine is provided', async () => {
      expect.assertions(1);

      const firstLine = 'firstLine';

      const element = (await mountAndWait(<SpaceListItem firstLine={firstLine} />))
        .find(`[data-type="body-primary"]`)
        .getDOMNode();

      expect(element.textContent).toBe(firstLine);
    });

    it('should have provided secondLine when secondLine is provided', async () => {
      expect.assertions(1);

      const secondLine = ['one', 'two'];

      const element = (
        await mountAndWait(<SpaceListItem firstLine="firstLine" secondLine={secondLine} />)
      )
        .find(`[data-type="body-secondary"]`)
        .getDOMNode();

      expect(element.textContent).toBe('one - two');
    });

    it('should have provided isNewActivity when isNewActivity is provided', async () => {
      expect.assertions(1);

      const isNewActivity = true;

      const element = (await mountAndWait(<SpaceListItem isNewActivity={isNewActivity} />))
        .find(Icon)
        .getDOMNode();

      expect(element).toBeDefined();
    });

    it('should have provided isMention when isMention is provided', async () => {
      expect.assertions(1);

      const isMention = true;

      const element = (await mountAndWait(<SpaceListItem isMention={isMention} />))
        .find(Icon)
        .getDOMNode();

      expect(element).toBeDefined();
    });

    it('should have provided isEnterRoom when isEnterRoom is provided', async () => {
      expect.assertions(1);

      const isEnterRoom = true;

      const element = (await mountAndWait(<SpaceListItem isEnterRoom={isEnterRoom} />))
        .find(Icon)
        .getDOMNode();

      expect(element).toBeDefined();
    });

    it('should have provided isAlertMuted when isAlertMuted is provided', async () => {
      expect.assertions(1);

      const isAlertMuted = true;

      const element = (await mountAndWait(<SpaceListItem isAlertMuted={isAlertMuted} />))
        .find(Icon)
        .getDOMNode();

      expect(element).toBeDefined();
    });

    it('should have provided isError when isError is provided', async () => {
      expect.assertions(1);

      const isError = true;

      const element = (await mountAndWait(<SpaceListItem isError={isError} />))
        .find(Icon)
        .getDOMNode();

      expect(element).toBeDefined();
    });

    it('should have provided isError when isError is provided', async () => {
      expect.assertions(1);

      const action = <p>action</p>;

      const element = (await mountAndWait(<SpaceListItem action={action} />))
        .find(`[data-position="end"]`)
        .getDOMNode();

      expect(element.textContent).toBe('action');
    });
  });
});
