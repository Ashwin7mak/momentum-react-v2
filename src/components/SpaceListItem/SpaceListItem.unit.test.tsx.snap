// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpaceListItem /> snapshot should match snapshot 1`] = `
<SpaceListItem>
  <ListItemBase
    className=""
    shape="isPilled"
    size={50}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="md-list-item-base-wrapper"
          data-disabled={false}
          data-padded={false}
          data-shape="isPilled"
          data-size={50}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
        >
          <ListItemBaseSection
            position="start"
          >
            <div
              data-position="start"
            />
          </ListItemBaseSection>
          <ListItemBaseSection
            className="md-space-list-item-text-wrapper"
            position="middle"
          >
            <div
              className="md-space-list-item-text-wrapper"
              data-position="middle"
            >
              <div>
                <Text
                  type="body-primary"
                >
                  <p
                    className="md-text-wrapper md-text-color"
                    data-type="body-primary"
                  />
                </Text>
              </div>
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="end"
          >
            <div
              data-position="end"
            />
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</SpaceListItem>
`;

exports[`<SpaceListItem /> snapshot should match snapshot with action 1`] = `
<SpaceListItem
  action={
    <p>
      action
    </p>
  }
>
  <ListItemBase
    className=""
    shape="isPilled"
    size={50}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="md-list-item-base-wrapper"
          data-disabled={false}
          data-padded={false}
          data-shape="isPilled"
          data-size={50}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
        >
          <ListItemBaseSection
            position="start"
          >
            <div
              data-position="start"
            />
          </ListItemBaseSection>
          <ListItemBaseSection
            className="md-space-list-item-text-wrapper"
            position="middle"
          >
            <div
              className="md-space-list-item-text-wrapper"
              data-position="middle"
            >
              <div>
                <Text
                  type="body-primary"
                >
                  <p
                    className="md-text-wrapper md-text-color"
                    data-type="body-primary"
                  />
                </Text>
              </div>
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="end"
          >
            <div
              data-position="end"
            >
              <p>
                action
              </p>
            </div>
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</SpaceListItem>
`;

exports[`<SpaceListItem /> snapshot should match snapshot with className 1`] = `
<SpaceListItem
  className="example-class"
>
  <ListItemBase
    className="example-class"
    shape="isPilled"
    size={50}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="example-class md-list-item-base-wrapper"
          data-disabled={false}
          data-padded={false}
          data-shape="isPilled"
          data-size={50}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
        >
          <ListItemBaseSection
            position="start"
          >
            <div
              data-position="start"
            />
          </ListItemBaseSection>
          <ListItemBaseSection
            className="md-space-list-item-text-wrapper"
            position="middle"
          >
            <div
              className="md-space-list-item-text-wrapper"
              data-position="middle"
            >
              <div>
                <Text
                  type="body-primary"
                >
                  <p
                    className="md-text-wrapper md-text-color"
                    data-type="body-primary"
                  />
                </Text>
              </div>
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="end"
          >
            <div
              data-position="end"
            />
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</SpaceListItem>
`;

exports[`<SpaceListItem /> snapshot should match snapshot with firstLine 1`] = `
<SpaceListItem
  firstLine="firstLine"
>
  <ListItemBase
    className=""
    shape="isPilled"
    size={50}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="md-list-item-base-wrapper"
          data-disabled={false}
          data-padded={false}
          data-shape="isPilled"
          data-size={50}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
        >
          <ListItemBaseSection
            position="start"
          >
            <div
              data-position="start"
            />
          </ListItemBaseSection>
          <ListItemBaseSection
            className="md-space-list-item-text-wrapper"
            position="middle"
          >
            <div
              className="md-space-list-item-text-wrapper"
              data-position="middle"
            >
              <div>
                <Text
                  type="body-primary"
                >
                  <p
                    className="md-text-wrapper md-text-color"
                    data-type="body-primary"
                  >
                    firstLine
                  </p>
                </Text>
              </div>
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="end"
          >
            <div
              data-position="end"
            />
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</SpaceListItem>
`;

exports[`<SpaceListItem /> snapshot should match snapshot with id 1`] = `
<SpaceListItem
  id="example-id"
>
  <ListItemBase
    className=""
    id="example-id"
    shape="isPilled"
    size={50}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="md-list-item-base-wrapper"
          data-disabled={false}
          data-padded={false}
          data-shape="isPilled"
          data-size={50}
          id="example-id"
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
        >
          <ListItemBaseSection
            position="start"
          >
            <div
              data-position="start"
            />
          </ListItemBaseSection>
          <ListItemBaseSection
            className="md-space-list-item-text-wrapper"
            position="middle"
          >
            <div
              className="md-space-list-item-text-wrapper"
              data-position="middle"
            >
              <div>
                <Text
                  type="body-primary"
                >
                  <p
                    className="md-text-wrapper md-text-color"
                    data-type="body-primary"
                  />
                </Text>
              </div>
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="end"
          >
            <div
              data-position="end"
            />
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</SpaceListItem>
`;

exports[`<SpaceListItem /> snapshot should match snapshot with isAlertMuted 1`] = `
<SpaceListItem
  isAlertMuted={true}
>
  <ListItemBase
    className=""
    shape="isPilled"
    size={50}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="md-list-item-base-wrapper"
          data-disabled={false}
          data-padded={false}
          data-shape="isPilled"
          data-size={50}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
        >
          <ListItemBaseSection
            position="start"
          >
            <div
              data-position="start"
            />
          </ListItemBaseSection>
          <ListItemBaseSection
            className="md-space-list-item-text-wrapper"
            position="middle"
          >
            <div
              className="md-space-list-item-text-wrapper"
              data-position="middle"
            >
              <div>
                <Text
                  type="body-primary"
                >
                  <p
                    className="md-text-wrapper md-text-color"
                    data-type="body-primary"
                  />
                </Text>
              </div>
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="end"
          >
            <div
              data-position="end"
            >
              <Icon
                fillColor="var(--listitem-icon)"
                name="alert-muted"
                scale={14}
                strokeColor="none"
                weight="bold"
              >
                <div
                  className="md-icon-wrapper"
                >
                  <svg
                    className=""
                    data-autoscale={false}
                    data-scale={14}
                    height="100%"
                    style={
                      Object {
                        "fill": "var(--listitem-icon)",
                        "stroke": "none",
                      }
                    }
                    viewBox="0, 0, 32, 32"
                    width="100%"
                  />
                </div>
              </Icon>
            </div>
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</SpaceListItem>
`;

exports[`<SpaceListItem /> snapshot should match snapshot with isEnterRoom 1`] = `
<SpaceListItem
  isEnterRoom={true}
>
  <ListItemBase
    className=""
    shape="isPilled"
    size={50}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="md-list-item-base-wrapper"
          data-disabled={false}
          data-padded={false}
          data-shape="isPilled"
          data-size={50}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
        >
          <ListItemBaseSection
            position="start"
          >
            <div
              data-position="start"
            />
          </ListItemBaseSection>
          <ListItemBaseSection
            className="md-space-list-item-text-wrapper"
            position="middle"
          >
            <div
              className="md-space-list-item-text-wrapper"
              data-position="middle"
            >
              <div>
                <Text
                  type="body-primary"
                >
                  <p
                    className="md-text-wrapper md-text-color"
                    data-type="body-primary"
                  />
                </Text>
              </div>
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="end"
          >
            <div
              data-position="end"
            >
              <Icon
                fillColor="var(--listitem-tick)"
                name="enter-room"
                scale={14}
                strokeColor="none"
                weight="bold"
              >
                <div
                  className="md-icon-wrapper"
                >
                  <svg
                    className=""
                    data-autoscale={false}
                    data-scale={14}
                    height="100%"
                    style={
                      Object {
                        "fill": "var(--listitem-tick)",
                        "stroke": "none",
                      }
                    }
                    viewBox="0, 0, 32, 32"
                    width="100%"
                  />
                </div>
              </Icon>
            </div>
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</SpaceListItem>
`;

exports[`<SpaceListItem /> snapshot should match snapshot with isError 1`] = `
<SpaceListItem
  isError={true}
>
  <ListItemBase
    className=""
    shape="isPilled"
    size={50}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="md-list-item-base-wrapper"
          data-disabled={false}
          data-padded={false}
          data-shape="isPilled"
          data-size={50}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
        >
          <ListItemBaseSection
            position="start"
          >
            <div
              data-position="start"
            />
          </ListItemBaseSection>
          <ListItemBaseSection
            className="md-space-list-item-text-wrapper"
            position="middle"
          >
            <div
              className="md-space-list-item-text-wrapper"
              data-position="middle"
            >
              <div>
                <Text
                  type="body-primary"
                >
                  <p
                    className="md-text-wrapper md-text-color"
                    data-type="body-primary"
                  />
                </Text>
              </div>
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="end"
          >
            <div
              data-position="end"
            >
              <Icon
                fillColor="var(--label-error-text)"
                name="priority-circle"
                scale={14}
                strokeColor="none"
                weight="filled"
              >
                <div
                  className="md-icon-wrapper"
                >
                  <svg
                    className=""
                    data-autoscale={false}
                    data-scale={14}
                    height="100%"
                    style={
                      Object {
                        "fill": "var(--label-error-text)",
                        "stroke": "none",
                      }
                    }
                    viewBox="0, 0, 32, 32"
                    width="100%"
                  />
                </div>
              </Icon>
            </div>
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</SpaceListItem>
`;

exports[`<SpaceListItem /> snapshot should match snapshot with isMention 1`] = `
<SpaceListItem
  isMention={true}
>
  <ListItemBase
    className=""
    shape="isPilled"
    size={50}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="md-list-item-base-wrapper"
          data-disabled={false}
          data-padded={false}
          data-shape="isPilled"
          data-size={50}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
        >
          <ListItemBaseSection
            position="start"
          >
            <div
              data-position="start"
            />
          </ListItemBaseSection>
          <ListItemBaseSection
            className="md-space-list-item-text-wrapper"
            position="middle"
          >
            <div
              className="md-space-list-item-text-wrapper"
              data-position="middle"
            >
              <div>
                <Text
                  type="body-primary"
                >
                  <p
                    className="md-text-wrapper md-text-color"
                    data-type="body-primary"
                  />
                </Text>
              </div>
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="end"
          >
            <div
              data-position="end"
            >
              <Icon
                fillColor="var(--listitem-tick)"
                name="mention"
                scale={14}
                strokeColor="none"
                weight="bold"
              >
                <div
                  className="md-icon-wrapper"
                >
                  <svg
                    className=""
                    data-autoscale={false}
                    data-scale={14}
                    height="100%"
                    style={
                      Object {
                        "fill": "var(--listitem-tick)",
                        "stroke": "none",
                      }
                    }
                    viewBox="0, 0, 32, 32"
                    width="100%"
                  />
                </div>
              </Icon>
            </div>
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</SpaceListItem>
`;

exports[`<SpaceListItem /> snapshot should match snapshot with isNewActivity 1`] = `
<SpaceListItem
  isNewActivity={true}
>
  <ListItemBase
    className="md-space-list-item-is-new-activity"
    shape="isPilled"
    size={50}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="md-space-list-item-is-new-activity md-list-item-base-wrapper"
          data-disabled={false}
          data-padded={false}
          data-shape="isPilled"
          data-size={50}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
        >
          <ListItemBaseSection
            position="start"
          >
            <div
              data-position="start"
            />
          </ListItemBaseSection>
          <ListItemBaseSection
            className="md-space-list-item-text-wrapper"
            position="middle"
          >
            <div
              className="md-space-list-item-text-wrapper"
              data-position="middle"
            >
              <div>
                <Text
                  type="body-primary"
                >
                  <p
                    className="md-text-wrapper md-text-color"
                    data-type="body-primary"
                  />
                </Text>
              </div>
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="end"
          >
            <div
              data-position="end"
            >
              <Icon
                fillColor="var(--listitem-tick)"
                name="unread"
                scale={14}
                strokeColor="none"
                weight="bold"
              >
                <div
                  className="md-icon-wrapper"
                >
                  <svg
                    className=""
                    data-autoscale={false}
                    data-scale={14}
                    height="100%"
                    style={
                      Object {
                        "fill": "var(--listitem-tick)",
                        "stroke": "none",
                      }
                    }
                    viewBox="0, 0, 32, 32"
                    width="100%"
                  />
                </div>
              </Icon>
            </div>
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</SpaceListItem>
`;

exports[`<SpaceListItem /> snapshot should match snapshot with secondLine 1`] = `
<SpaceListItem
  secondLine="secondLine"
>
  <ListItemBase
    className=""
    shape="isPilled"
    size={50}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="md-list-item-base-wrapper"
          data-disabled={false}
          data-padded={false}
          data-shape="isPilled"
          data-size={50}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
        >
          <ListItemBaseSection
            position="start"
          >
            <div
              data-position="start"
            />
          </ListItemBaseSection>
          <ListItemBaseSection
            className="md-space-list-item-text-wrapper"
            position="middle"
          >
            <div
              className="md-space-list-item-text-wrapper"
              data-position="middle"
            >
              <div>
                <Text
                  type="body-primary"
                >
                  <p
                    className="md-text-wrapper md-text-color"
                    data-type="body-primary"
                  />
                </Text>
                <Text
                  style={
                    Object {
                      "color": "var(--theme-text-team-default-normal)",
                    }
                  }
                  type="body-secondary"
                >
                  <small
                    className="md-text-wrapper md-text-color"
                    data-type="body-secondary"
                    style={
                      Object {
                        "color": "var(--theme-text-team-default-normal)",
                      }
                    }
                  >
                    secondLine
                  </small>
                </Text>
              </div>
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="end"
          >
            <div
              data-position="end"
            />
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</SpaceListItem>
`;

exports[`<SpaceListItem /> snapshot should match snapshot with style 1`] = `
<SpaceListItem
  style={
    Object {
      "color": "pink",
    }
  }
>
  <ListItemBase
    className=""
    shape="isPilled"
    size={50}
    style={
      Object {
        "color": "pink",
      }
    }
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="md-list-item-base-wrapper"
          data-disabled={false}
          data-padded={false}
          data-shape="isPilled"
          data-size={50}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
          style={
            Object {
              "color": "pink",
            }
          }
        >
          <ListItemBaseSection
            position="start"
          >
            <div
              data-position="start"
            />
          </ListItemBaseSection>
          <ListItemBaseSection
            className="md-space-list-item-text-wrapper"
            position="middle"
          >
            <div
              className="md-space-list-item-text-wrapper"
              data-position="middle"
            >
              <div>
                <Text
                  type="body-primary"
                >
                  <p
                    className="md-text-wrapper md-text-color"
                    data-type="body-primary"
                  />
                </Text>
              </div>
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="end"
          >
            <div
              data-position="end"
            />
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</SpaceListItem>
`;

exports[`<SpaceListItem /> snapshot should match snapshot with teamColor 1`] = `
<SpaceListItem
  teamColor="cobalt"
>
  <ListItemBase
    className=""
    shape="isPilled"
    size={50}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="md-list-item-base-wrapper"
          data-disabled={false}
          data-padded={false}
          data-shape="isPilled"
          data-size={50}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
        >
          <ListItemBaseSection
            position="start"
          >
            <div
              data-position="start"
            />
          </ListItemBaseSection>
          <ListItemBaseSection
            className="md-space-list-item-text-wrapper"
            position="middle"
          >
            <div
              className="md-space-list-item-text-wrapper"
              data-position="middle"
            >
              <div>
                <Text
                  type="body-primary"
                >
                  <p
                    className="md-text-wrapper md-text-color"
                    data-type="body-primary"
                  />
                </Text>
              </div>
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="end"
          >
            <div
              data-position="end"
            />
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</SpaceListItem>
`;
